### ISSUE TITLE
End resolvers - ideas?

### ISSUE THREAD

**Issue by RatkoR** [99designs/gqlgen#2613]
Hi all, I have an interestng problem that I cannot easily solve. I'll use the example from gqlgen to explain it.

We have list of TODOs, users are loaded on-demand (see [here](https://gqlgen.com/getting-started/#dont-eagerly-fetch-the-user)). In our workflow, we always load users concurrently with dataloaders, it's super useful.

Let's say that these users have also a status flag and we don't want to show TODOs of users that are disabled.

How can we do that?

The problem is that we can handle and change TODO slice in `Todos(ctx context.Context)` before TODOs are returned to gqlgen. But we know nothing about Users there yet. gqlgen takes those TODOs, glues User to every one and returns the result. We could load User data in Todos() function, but this would load them seqentially and not concurrently like dataloader does.

If gqlgen had an end resolver (of final resolver, or whatever name) which would be run after field resolvers are done, we could then filter TODOs based on those User objects.

----

This would tell gqlgen to create an "end" resolver:

```
models:
  ID:
    ...
  Todo:
    fields:
      user:
        resolver: true
    resolver: true    // <-- this
```
an generate a method something like:

```
func (r *todoResolver) End(ctx context.Context, obj []*model.Todo) ([]*model.Todo, error) {
        // do whatever we need with TODO and it's User
	return obj, nil
}
```

Thoughts?

**Comment by collier** [comment]
@RatkoR I'd just like to add I am facing the exact same situation when it comes to needing a way to filter the results of a dataloader based on some query input the loader can't access. Did you ever find a workaround solution for your particular problem?

**Comment by RatkoR** [comment]
No, unfortunately not. Instead of loading users with dataloaders, we load them together with todo items.
